api/grpcserver/node_service.go

```diff
@@ -123,7 +125,11 @@ func (s NodeService) Shutdown(context.Context, *pb.ShutdownRequest) (*pb.Shutdow
 // StatusStream exposes a stream of node status updates
 func (s NodeService) StatusStream(_ *pb.StatusStreamRequest, stream pb.NodeService_StatusStreamServer) error {
 	log.Info("GRPC NodeService.StatusStream")
-	statusStream := events.GetStatusChannel()
+	var statusStream chan struct{}
+	if s.statusChannel == nil {
+		statusStream = events.SubscribeToStatus(30)
```

Comment:
Please turn this hardcoded number 30 into a param. You might want the default value to live in the reporter rather than in the API, or to be used across the API.